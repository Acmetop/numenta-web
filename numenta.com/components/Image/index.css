/* Numenta company website source code. Copyright Â© 2016 Numenta.*/
/* Full details in LICENSE.txt, or contact us at <http://numenta.com>.*/
/* This program is free software: you can redistribute it and/or modify it under*/
/* the terms of the GNU Affero General Public License as published by the Free*/
/* Software Foundation, either version 3 of the License, or (at your option) any*/
/* later version. This program is distributed in the hope that it will be*/
/* useful, but WITHOUT ANY WARRANTY; without even the implied warranty of*/
/* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero*/
/* General Public License for more details. You should have received a copy of*/
/* the GNU Affero General Public License along with this program. If not, see*/
/* <https://www.gnu.org/licenses/agpl.html>.*/

.mw {
  composes: mw-100 from 'tachyons';
}
.w {
  composes: w-100 from 'tachyons';
}


.center {
  composes: center from 'tachyons';
}
.border {
  composes: b--black-10 from 'tachyons';
  composes: ba from 'tachyons';
}
.round {
  composes: br-100 from 'tachyons';
}
.shadow {
  composes: shadow-4 from 'tachyons';
}


.image {
  composes: br2 from 'tachyons';
  composes: mb1 from 'tachyons';
}
