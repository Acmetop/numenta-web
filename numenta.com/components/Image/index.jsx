import classNames from 'classnames'
import {prefixLink} from 'gatsby-helpers'  // eslint-disable-line import/no-unresolved, max-len
import React from 'react'

import styles from './index.css'


const Image = ({alt, border, onClick, respond, shadow, src, title}) => {
  const classes = [styles.image]
  const prefix = prefixLink()
  let source = src

  if (prefix && source && !source.match(prefix)) {
    source = prefixLink(source)
  }

  if (border) classes.push(styles.border)
  if (respond) classes.push(styles[respond])
  if (shadow) classes.push(styles.shadow)

  return (
    <img
      alt={alt}
      className={classNames(...classes)}
      onClick={onClick}
      src={source}
      title={title}
    />
  )
}

Image.propTypes = {
  alt: React.PropTypes.string.isRequired,
  border: React.PropTypes.bool,
  onClick: React.PropTypes.func,
  respond: React.PropTypes.oneOf(['mw', 'w']),
  shadow: React.PropTypes.bool,
  src: React.PropTypes.string.isRequired,
  title: React.PropTypes.string,
}

Image.defaultProps = {
  border: false,
  shadow: true,
}

export default Image
