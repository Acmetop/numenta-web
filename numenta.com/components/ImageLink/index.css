/**
 * Numenta.com company website source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2016 Numenta <http://numenta.com>
 */

._visited:visited {
  color: #1aa0db;
}


.imagelink {
  composes: dim from 'tachyons';
  composes: link from 'tachyons';
  composes: color-blue from '../../static/assets/css/theme.css';
  composes: _visited;
}
