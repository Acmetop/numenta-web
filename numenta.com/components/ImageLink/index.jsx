import {IndexLink, Link} from 'react-router'
import {prefixLink} from 'gatsby-helpers'  // eslint-disable-line import/no-unresolved, max-len
import React from 'react'

import styles from './index.css'


/**
 *
 */
const ImageLink = ({children, title, to}) => {
  const attrs = {
    // default internal non-index link
    className: styles.imagelink,
    title,
    to: prefixLink(to),
  }
  let Node = Link

  if (to.match(/^.+:/)) {
    // external link (browser location)
    Node = 'a'
    attrs.href = to
    delete attrs.to
  }
  else if (to === '/') {
    // internal index main/home link
    Node = IndexLink
  }

  return (
    <Node {...attrs}>
      {children}
    </Node>
  )
}

ImageLink.propTypes = {
  children: React.PropTypes.any.isRequired,
  title: React.PropTypes.string,
  to: React.PropTypes.string.isRequired,
}

export default ImageLink
