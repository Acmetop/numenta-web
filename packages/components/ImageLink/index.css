/**
 * Numenta shared web components
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2016 Numenta <http://numenta.com>
 */

._visited:visited {
  color: #1aa0db;
}


.imagelink {
  composes: dim from 'tachyons';
  composes: link from 'tachyons';
  composes: _visited;
  color: #1aa0db;
}
