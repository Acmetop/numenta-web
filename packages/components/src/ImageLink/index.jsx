// Numenta Web Platform and Sites source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2017 Numenta <http://numenta.com>

import {IndexLink, Link} from 'react-router'
import {prefixLink} from 'gatsby-helpers'
import React from 'react'

import {triggerGAnalyticsEvent} from 'numenta-web-shared-utils/lib/client'

import styles from './index.css'


/**
 *
 */
const ImageLink = ({children, title, itemProp, to}) => {
  const onClick = (event) => {
    // get focus off clicked image
    event.target.parentNode.blur()
    // send ga event for asset link/download
    return triggerGAnalyticsEvent(event.target.getAttribute('href'))
  }
  const attrs = {
    // default internal non-index link
    className: styles.imagelink,
    onClick,
    title,
    itemProp,
    to: prefixLink(to),
  }
  let Node = Link

  if (to && (
    to.match(/^.+:/) || to.match(/^\/assets\//) || to.match(/\.pdf$/)
  )) {
    // external link (browser location)
    Node = 'a'
    if (to.match(/^.*:/)) {
      attrs.href = to  // = mailto:etc@blah.com
    }
    else if (to.match(/^\/assets\//) || to.match(/\.pdf$/)) {
      attrs.href = prefixLink(to)  // = /assets/etc/
    }
    delete attrs.to
  }
  else if (to === '/') {
    // internal index main/home link
    Node = IndexLink
  }

  return (
    <Node {...attrs}>
      {children}
    </Node>
  )
}

ImageLink.propTypes = {
  children: React.PropTypes.node.isRequired,
  title: React.PropTypes.string,
  itemProp: React.PropTypes.string,
  to: React.PropTypes.string.isRequired,
}

export default ImageLink
