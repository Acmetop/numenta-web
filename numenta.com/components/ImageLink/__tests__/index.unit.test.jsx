import React from 'react'
import renderer from 'react-test-renderer'

import Image from '../../Image'
import ImageLink from '../../ImageLink'


describe('ImageLink React component', () => {

  it('Renders correctly', () => {
    const component = renderer.create(
      <ImageLink to="http://numenta.com">
        <Image alt="Zebra" src="zebra.png" />
      </ImageLink>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
