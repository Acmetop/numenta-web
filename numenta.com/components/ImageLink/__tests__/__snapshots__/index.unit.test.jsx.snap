exports[`ImageLink React component Renders correctly 1`] = `
<a
  className="imagelink"
  href="http://numenta.com"
  onClick={[Function onClick]}
  title={undefined}>
  <img
    alt="Zebra"
    className="image shadow"
    onClick={undefined}
    src="zebra.png"
    title={undefined} />
</a>
`;
